modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1290,15 +1290,23 @@ void GDScriptAnalyzer::resolve_class_body(GDScriptParser::ClassNode *p_class, co
 							return_datatype.is_meta_type = false;
 						}
 
-						if (getter_function->parameters.size() != 0 || return_datatype.has_no_type()) {
+						if (getter_function->mandatory_parameter_count > 1 || return_datatype.has_no_type()) {
 							push_error(vformat(R"(Function "%s" cannot be used as getter because of its signature.)", getter_function->identifier->name), member.variable);
 						} else if (!is_type_compatible(member.variable->datatype, return_datatype, true)) {
 							push_error(vformat(R"(Function with return type "%s" cannot be used as getter for a property of type "%s".)", return_datatype.to_string(), member.variable->datatype.to_string()), member.variable);
-
 						} else {
 							has_valid_getter = true;
+							if (getter_function->mandatory_parameter_count == 1) {
+								GDScriptParser::DataType name_arg_type = getter_function->parameters[0]->get_datatype();
+								if (!name_arg_type.is_variant() && name_arg_type.is_hard_type()) {
+									if (name_arg_type.kind != GDScriptParser::DataType::BUILTIN || (name_arg_type.builtin_type != Variant::STRING && name_arg_type.builtin_type != Variant::STRING_NAME)) {
+										push_error(vformat(R"(Function with name argument of type "%s" cannot be used as getter for a property.)", name_arg_type.to_string()), member.variable);
```

Comment:
I think it might be best shorter, not mentioning the signature but just the issue, unsure about how to word it but this gets pretty long

Reply:
You're right, the error message could be better.

How about "First argument of a getter "%s" should be a property name and must be a String or StringName"?

Reply:
The problem is that 1 argument does **not** have to be a String or StringNamr, generally speaking. It only has to be a string if the mandatory argument count is 2 (and also you can choose not to provide a type hint at all). Otherwise, the first argument of the answer is the "value argument", not the "name argument".

I have used these terms so as not to bring all these details to the user in a short error message, in the assumption that these terms will be used in online documentation.